llvm/unittests/IR/IntrinsicsTest.cpp

```diff
@@ -74,15 +74,74 @@ TEST(IntrinsicNameLookup, Basic) {
   };
 
   for (const auto &[Name, ExpectedIdx] : Tests) {
-    int Idx = Intrinsic::lookupLLVMIntrinsicByName(NameTable1, Name);
+    int Idx = Intrinsic::lookupLLVMIntrinsicByName(NameTable, Name);
     EXPECT_EQ(ExpectedIdx, Idx);
     if (!StringRef(Name).starts_with("llvm.foo"))
       continue;
-    Idx = Intrinsic::lookupLLVMIntrinsicByName(NameTable1, Name, "foo");
+    Idx = Intrinsic::lookupLLVMIntrinsicByName(NameTable, Name, "foo");
     EXPECT_EQ(ExpectedIdx, Idx);
   }
 }
 
+// Test case to demonstrate potential conflicts between overloaded and non-
+// overloaded intrinsics. The name match works by essentially dividing then
+// name into . separated components and doing successive search for each
+// component. When a search fails, the lowest component of the matching
+// range for the previous component is returned.
+TEST(IntrinsicNameLookup, OverloadConflict) {
```

Comment:
I'm not sure I understand the purpose of this test. Why are we testing a situation that is forbidden by tablegen?

Reply:
I guess it was used to demonstrate/discover the conflict conditions initially be me and I thought it might be useful to keep it as a test case to demonstrate the condition under which the lookup fails in the presence of conflicts. I agree that it will be forbidden now, so maybe there is no value in preserving it. I can delete it if you suggest so.

Reply:
I'd drop it, but I won't insist if you think it's valuable.

Reply:
It has served its purpose, so will drop it.

Reply:
Done.